util/url/url.go

```diff
@@ -0,0 +1,16 @@
+package url
+
+import "net/url"
+
+// AppendPortIfMissing takes a URL and port as strings and appends the provided
+// port to the URL if it is missing
+func AppendPortIfMissing(u string, p string) (string, error) {
+	parsedURL, err := url.Parse(u)
+	if err != nil {
+		return "", err
+	}
+	if parsedURL.Port() == "" {
+		return parsedURL.String() + ":" + p, nil
```

Comment:
If the parsed URL contained a path it could cause an issue, might be more safe to do:

```go
parsedURL.Host = parsedURL.Hostname() + ":" + p
return parsedURL.String(), nil
```